src/globe/Globe.js

```diff
@@ -659,6 +659,38 @@ define([
             return this.elevationModel.elevationsForGrid(sector, numLat, numLon, targetResolution, result);
         };
 
+        /**
+         * Computes a new set of locations translated from a specified location to a new location for this globe.
+         *
+         * @param {Location} oldLocation The original reference location.
+         * @param {Location} newLocation The new reference location.
+         * @param {Location[]} locations The locations to translate.
+         *
+         * @return {Location[]} The translated locations.
+         */
+        Globe.prototype.computeShiftedLocations = function(oldLocation, newLocation, locations) {
+            var newLocations = [];
+
+            var oldPoint = this.computePointFromLocation(oldLocation.latitude, oldLocation.longitude,
+                new Vec3(0, 0, 0));
+            var newPoint = this.computePointFromLocation(newLocation.latitude, newLocation.longitude,
+                new Vec3(0, 0, 0));
+            var delta = newPoint.subtract(oldPoint);
+
+            for(var i =0; i < locations.length; i++)
+            {
+                var point = this.computePointFromLocation(locations[i].latitude, locations[i].longitude,
+                    new Vec3(0, 0, 0));
```

Comment:
This Vec3 could be declared before the for loop and reused to save some object allocations.